### PR TITLE
Staking: `chill_other` takes stash instead of controller

### PR DESCRIPTION
**Pull request by rossbulat**
The `chill_other` call is the only staking call that explicitly requires `controller` in its signature. This PR changes the controller arg to be the stash instead, with `StakingLedger` then fetching the controller from storage.

This is not a breaking change per se - the call types do not change, but is noteworthy as UIs will now want to pass the stash account into `chill_other` calls, & metadata will reflect this.

Note: This is very low impact. `chill_other` has [hardly ever been used](https://polkadot.subscan.io/extrinsic?address=&module=staking&call=chill_other&result=all&signedChecked=signed%20only&startDate=&endDate=&startBlock=&timeType=date&version=9431&endBlock=) on Polkadot - notwithstanding the one called 11 days ago at block 18177457 that was a part of test I did, the last call was made 493 days ago. Only 2 calls have ever been successful.

Addresses controller deprecation #2500

### CODE REVIEW

**Comment by rossbulat** [comment]
bot bench substrate-pallet --pallet=pallet_staking

**Review by Ank4n — APPROVED** [review]

File: substrate/frame/staking/src/pallet/mod.rs [hunk 1]
```diff
@@ -1327,7 +1327,7 @@ pub mod pallet {
 			// (temporary) passive migration.
 			Self::ledger(StakingAccount::Stash(stash.clone())).map(|ledger| {
 				let controller = ledger.controller()
-                    .defensive_proof("ledger was fetched used the StakingInterface, so controller field must exist; qed.")
+                    .defensive_proof("ledger was fetched using StakingLedger, so controller field must exist; qed.")
```
Comment (gpestana) [inline substrate/frame/staking/src/pallet/mod.rs:1330]: 👍

File: substrate/frame/staking/src/pallet/mod.rs [hunk 2]
```diff
@@ -1764,11 +1764,16 @@ pub mod pallet {
...
+		pub fn chill_other(origin: OriginFor<T>, stash: T::AccountId) -> DispatchResult {
 			// Anyone can call this function.
 			let caller = ensure_signed(origin)?;
-			let ledger = Self::ledger(Controller(controller.clone()))?;
-			let stash = ledger.stash;
+			let ledger = Self::ledger(Stash(stash.clone()))?;
+			let controller = ledger
+				.controller()
+				.defensive_proof(
+					"ledger was fetched using StakingLedger, so controller field must exist; qed.",
```
Comment (gpestana) [inline substrate/frame/staking/src/pallet/mod.rs:1774]:
```suggestion
					"Ledger's controller field didn't exist. The ledger should fetched using StakingLedger.",
```

**Review by gpestana — APPROVED** [review]

File: substrate/frame/staking/src/pallet/mod.rs (hunk 2 above)
Comment (gpestana) [reply substrate/frame/staking/src/pallet/mod.rs:1774]: `qed` is more for `expect()`s :)

**Pull request merged**